### PR TITLE
Extract getContributionBalance function, use that rather than wrapper…

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code cleanup to extract the guts of an overloaded function so just the relevant part can be called. These are internal functions only.

Before
----------------------------------------
Complex function ``function getPartialPaymentWithType`` doing multiple things

After
----------------------------------------
Get balance part callable by itself

Technical Details
----------------------------------------
@monishdeb @pradpnayak this function is only called from a few places and I couldn't find any evidence that the 'full_paid' parameter is ever used so I dropped that. I feel like this 'getByType' stuff is a bit silly & we should determine that before calling it but for now I haven't touched that.

I also updated the payment only bit from the contribution form to call this function - this came off trying to understand https://github.com/civicrm/civicrm-core/pull/12319/files#diff-e59ea24da123edeec35a0c498d02874fR1322 & feeling that we just need to know the amount owing whatever the status

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -4168,6 +4168,39 @@ public static function getPaymentInfo($id, $component, $getTrxnInfo = FALSE, $us
...
+
+    $sqlFtTotalAmt = "
+SELECT SUM(ft.total_amount)
+FROM civicrm_financial_trxn ft
+  INNER JOIN civicrm_entity_financial_trxn eft ON (ft.id = eft.financial_trxn_id AND eft.entity_table = 'civicrm_contribution' AND eft.entity_id = {$contributionId})
+WHERE ft.is_payment = 1
+  AND ft.status_id IN ({$statusId}, {$refundStatusId})
+";
+
+    $ftTotalAmt = CRM_Core_DAO::singleValueQuery($sqlFtTotalAmt);
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: @monishdeb ok - wanna do that as a follow up?

**Comment by eileenmcnaughton** [comment]
@monishdeb can we merge this?

File: CRM/Core/BAO/FinancialTrxn.php [hunk 2]
```diff
@@ -494,9 +478,6 @@ public static function getPartialPaymentWithType($entityId, $entityName = 'parti
         elseif ($paymentVal > 0) {
           $value['amount_owed'] = $paymentVal;
         }
-        elseif ($lineItemTotal == $ftTotalAmt) {
-          $value['full_paid'] = $ftTotalAmt;
-        }
```
Comment (monishdeb) [inline CRM/Core/BAO/FinancialTrxn.php]: lol and I didn't relaised it's not used anwhere in Civi but still we are having a separate attribute.
```
$ grep -irn "full_paid" CRM/
CRM//Core/BAO/FinancialTrxn.php:498:          $value['full_paid'] = $ftTotalAmt;
```
Agree with this change

File: tests/phpunit/CRM/Event/BAO/AdditionalPaymentTest.php [hunk 3]
```diff
@@ -207,16 +207,15 @@ public function testAddPartialPayment() {
     $amtPaid = 60;
     $balance = $feeAmt - $amtPaid;
     $result = $this->addParticipantWithPayment($feeAmt, $amtPaid);
-    extract($result);
-    $paymentInfo = CRM_Contribute_BAO_Contribution::getPaymentInfo($participant['id'], 'event');
+    $paymentInfo = CRM_Contribute_BAO_Contribution::getPaymentInfo($result['participant']['id'], 'event');
```
Comment (monishdeb) [inline tests/phpunit/CRM/Event/BAO/AdditionalPaymentTest.php]: better :)

**Comment by monishdeb** [comment]
Agree with the overall change and existing UTs ensure no fee-change is broken by this patch. Though there's a scope for optimization that I'll will bring in a separate followup PR.

* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __PASS__
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __PASS__
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __PASS__
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __PASS__
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __PASS__
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))__PASS__
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __PASS__
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __PASS__

File: CRM/Contribute/BAO/Contribution.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (monishdeb) [reply CRM/Contribute/BAO/Contribution.php]: Submitted followup PR https://github.com/civicrm/civicrm-core/pull/13187

**Pull request merged**